sunpy/database/tables.py

```diff
@@ -374,42 +376,53 @@ def _from_fido_search_result_block(cls, sr_block, default_waveunit=None):
-        source = sr_block.get('Source', None)
-        provider = sr_block.get('Provider', None)
-        physobs = sr_block.get('Physobs', None)
+        source = sr_block.get('Source')
+        provider = sr_block.get('Provider')
+        physobs = sr_block.get('Physobs')
         if physobs is not None:
             physobs = str(physobs)
-        instrument = sr_block.get('Instrument', None)
-        time_start = sr_block['Time'].start.datetime
-        time_end = sr_block['Time'].end.datetime
+        instrument = sr_block.get('Instrument')
+        time_start = sr_block.get('Start Time')
+        if time_start is not None:
+            time_start = time_start.datetime
+        time_end = sr_block.get('End Time')
+        if time_end is not None:
+            time_end = time_end.datetime
 
         wavelengths = sr_block.get('Wavelength', np.nan)
-        wavelength_temp = {}
-        if isinstance(wavelength_temp, tuple):
-            # Tuple of values
-            wavelength_temp['wavemin'] = wavelengths[0]
-            wavelength_temp['wavemax'] = wavelengths[1]
+        if isinstance(wavelengths, u.Quantity):
+            if wavelengths.isscalar:
+                wavemin = wavemax = wavelengths.to_value(u.nm, equivalencies.spectral())
+            else:
+                wavemin, wavemax = wavelengths.to_value(u.nm, equivalencies.spectral())
         else:
-            # Single Value
-            wavelength_temp['wavemin'] = wavelength_temp['wavemax'] = wavelengths
-
-        final_values = {}
-        for key, val in wavelength_temp.items():
-            if isinstance(val, quantity.Quantity):
-                unit = getattr(val, 'unit', None)
-                if unit is None:
-                    if default_waveunit is not None:
-                        unit = Unit(default_waveunit)
-                    else:
-                        raise WaveunitNotFoundError(sr_block)
-                final_values[key] = unit.to(nm, float(val.value), equivalencies.spectral())
-            elif val is None or np.isnan(val):
-                final_values[key] = val
-
-        wavemin = final_values['wavemin']
-        wavemax = final_values['wavemax']
-
-        fileid = sr_block.get('url', None)
+            wavelength_temp = {}
+            if isinstance(wavelength_temp, tuple):
+                # Tuple of values
+                wavelength_temp['wavemin'] = wavelengths[0]
+                wavelength_temp['wavemax'] = wavelengths[1]
+            else:
+                # Single Value
+                wavelength_temp['wavemin'] = wavelength_temp['wavemax'] = wavelengths
+
+            final_values = {}
+            for key, val in wavelength_temp.items():
+                if isinstance(val, quantity.Quantity):
+                    unit = getattr(val, 'unit', None)
+                    if unit is None:
+                        if default_waveunit is not None:
+                            unit = u.Unit(default_waveunit)
```

Comment:
These uncovered lines should get tests

Reply:
I am wondering if this whole else block can go away...

Reply:
I removed it, the new code path is all we need.